gateway/handle_http_auth.go

```diff
@@ -119,6 +121,45 @@ func (gw *Handle) sourceIDAuth(delegate http.HandlerFunc) http.HandlerFunc {
 	}
 }
 
+// authenticateDestinationIDForSource middleware to authenticate destinationId in the X-Rudder-Destination-Id header.
+// If the destinationId is invalid, the request is rejected.
+// destinationID authentication should be performed only after source is authenticated and source is present in context
+// Following validations are performed
+//  1. Destination should be present for source config
+//  2. Destination should be enabled for the source
+func (gw *Handle) authenticateDestinationIDForSource(delegate http.HandlerFunc) http.HandlerFunc {
```

Comment:
please consider picking up a shorter name :)
```suggestion
func (gw *Handle) sourceDestIDAuth(delegate http.HandlerFunc) http.HandlerFunc {
```
Additionally, since the `sourceIDAuth` is a prerequisite for this middleware, maybe it should embed it here, see `replaySourceIDAuth` for a similar example